### PR TITLE
executor: grant statement would create user if not exist

### PR DESCRIPTION
**Pull request by tiancaiamao**
```
grant all privileges on *.* to 'xxx'@'%' identified by 'yyy'
```

if 'xxx'@'%' doesn't exist, it will create a new user, rather than report error.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@tiancaiamao Please resolve the conflict.

File: executor/grant_test.go [hunk 1]
```diff
@@ -188,6 +188,16 @@ func (s *testSuite) TestIssue2456(c *C) {
...
 }
 
+func (s *testSuite) TestCreateUserWhenGrant(c *C) {
+	defer testleak.AfterTest(c)()
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("DROP USER IF EXISTS 'test'@'%'")
+	tk.MustExec("GRANT ALL PRIVILEGES ON *.* to 'test'@'%' IDENTIFIED BY 'xxx'")
+	// Make sure user is created automaticly when grant to a non-exists one.
+	rows := tk.MustQuery("SELECT user FROM mysql.user WHERE user='test' and host='%'").Rows()
+	c.Assert(rows, HasLen, 1)
```
Comment (zimulala) [inline executor/grant_test.go]: Check the value of rows.

**Comment by tiancaiamao** [comment]
PTAL @zimulala

**Comment by zimulala** [comment]
LGTM

**Pull request merged**